.github/workflows/python-scan-sbom.yml

```diff
@@ -0,0 +1,54 @@
+name: Downloads an artifact of a given name and scan the SBOM
+
+on:
+  workflow_call:
+    inputs:
+      SCAN_NAME:
+        type: string
+        description: |
+          The name of the scan to download the artifact of to scan
+        required: true
+
+jobs:
+  scan-sbom-for-high-crit:
+    name: Scan SBOM for HIGH and CRITICAL
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download SBOM
+        uses: actions/download-artifact@v4
+        with:
+          name: ${{ inputs.SCAN_NAME}}
+      - name: Scan SBOM
+        uses: aquasecurity/trivy-action@master
+        with:
+          exit-code: '1'
+          format: 'table'
+          severity: 'HIGH,CRITICAL'
+          scan-ref: ${{ inputs.SCAN_NAME }}-bom.json
+          scan-type: 'sbom'
+          scanners: 'vuln'
+
+  scan-and-store-full-report:
+    name: Scan for all vulnerabilities and store the report
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download SBOM
```

Comment:
Bit of a trivial observation... I'm not certain of the behaviour here. We already download the same artifact in line 17. Would this download overwrite this file or is it given a different name eg ..(1)? Are we actually scanning the file downloaded earlier?
If we don't care which downloaded artifact we are scanning, because they are the same file, do we need to download twice?